### ISSUE TITLE
New Rally-X initial screen upside down in cocktail mode

### ISSUE THREAD

**Issue by rdagger** [libretro/mame2003-plus-libretro#1571]
If you set the dip switch in New Rally-X to Cocktail mode, the screen flips upside down for player 1. Then, when player 2 starts, the screen flips to the orientation that player 1 should have seen originally.

**Comment by mahoneyt944** [comment]
Might be a video_update issue with rallyx, oddly other games don't seem effected

**Comment by rdagger** [comment]
I verified the hashes and tested the following:

1. Rally X (32k Ver.?) -> rallyx
2. Rally X (Midway) -> rallyxm
3. New Rally X -> nrallyx

They all have the correct orientation when cabinet is set to upright but are upside down when set to cocktail.

**Comment by mahoneyt944** [comment]
I'm sure I could hack a fix in for this, but I haven't found why only rallyx and it's variants have this issue. Other drivers using this flip write are fine. Curious if this was an anomaly in the layout of the  actual cocktail table or some sort of long standing issue with rallyx video update.

**Comment by rdagger** [comment]
Would it be possible to add a Dip Switches option to flip screen?

**Comment by mahoneyt944** [comment]
You can override the rotation in RetroArch.

Settings-> user interface-> show advanced settings-> on

Then

Settings-> Video-> output-> video rotation-> 180 deg

**Comment by rdagger** [comment]
I'm using multiple versions of Batocera so it took me a few hours to figure out the rotation on a per game basis.
With the override, both Rally X and New Rally X play correctly in cocktail mode.  I assume the midway version works too.

One issue is that if you press the "Tab" key to open the MAME options, the options screen will now be upside down.  So, I'd consider this a hack fix.

In case anyone else runs into the same problem, you can create a rallyx.cfg file containing:
`video_rotation = "2"`
Then place the .cfg file in:
`\share\system\.config\retroarch\config\MAME 2003-Plus`
for the 32-bit version of Batocera (v.30)
or
`\share\system\configs\retroarch\config\MAME 2003-Plus`
for the 64-bit version of Batocera (v.36)

I confirmed the overrides persist system reboots.

**Comment by mahoneyt944** [comment]
Yeah probably the best we can do, I believe the monitor in the cocktail table is physically rotated 180 degrees from that of the upright cabinet. So by fixing one we'd break the other.
